core-plugin/src/main/groovy/com/github/jrubygradle/api/core/RepositoryHandlerExtension.groovy

```diff
@@ -0,0 +1,116 @@
+package com.github.jrubygradle.api.core
+
+import com.github.jrubygradle.api.gems.GemGroups
+import com.github.jrubygradle.internal.core.IvyXmlGlobalProxyRegistry
+import groovy.transform.CompileDynamic
+import groovy.transform.CompileStatic
+import org.gradle.api.Action
+import org.gradle.api.Project
+import org.gradle.api.artifacts.repositories.ArtifactRepository
+import org.gradle.api.artifacts.repositories.IvyArtifactRepository
+import org.gradle.api.artifacts.repositories.MavenArtifactRepository
+import org.gradle.util.GradleVersion
+
+/** Extension which can be added to {@code project.repositories}.
+ *
+ * @author Schalk W. Cronjé
+ *
+ * @since 2.0
+ */
+@CompileStatic
+class RepositoryHandlerExtension {
+    public static final String NAME = 'ruby'
+    public static final String DEFAULT_GROUP_NAME = 'rubygems'
+
+    /** Creates an extension and associate it with a project.
+     *
+     * @param project Gradle project.
+     */
+    RepositoryHandlerExtension(final Project project) {
+        this.project = project
+        this.ivyProxies = new IvyXmlGlobalProxyRegistry(project)
```

Comment:
A registry should (theoretically) be a single instance across all of the project instances in a multi-project.